### PR TITLE
docs: update rbac docs

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

integrate the new machine-to-machine role into one doc

### CODE REVIEW

File: docs/docs/recipes/rbac/manage-permissions-and-roles.md [hunk 1]
```diff
@@ -28,9 +28,16 @@ To create a new role, click the "Create Role" button in the top right corner. A
 
 1. Keep in mind that while it is technically possible to create a role _without_ permissions or users assigned, it is not recommended to create too many empty roles. This will disrupt the harmony of role management and render the RBAC system ineffective.
 2. Permissions are grouped by API in the selector, allowing you to add them in bulk or select them individually. The selected permissions will be displayed on the right side.
-3. After creating a role, you will be taken to the user assignment page, where you can authorize the role to specific users. You can search for users by name, email, phone number, username or ID if available. If you prefer, you can also skip this step and assign users to the role later.
+3. A new role has two available types: user role or machine-to-machine app role.
```
Comment (darcyYe) [inline docs/docs/recipes/rbac/manage-permissions-and-roles.md:31]: nit: not sure user and machine-to-machine should be wrapped by quotes.

File: docs/docs/recipes/rbac/manage-permissions-and-roles.md [hunk 2]
```diff
@@ -39,24 +46,16 @@ After completing the role creation and assignment process or click a role name i
...
-![Manage Role Users](./assets/manage-role-users.png)
-
-Search or browse for the desired users and click "Assign Users" to add them to the role.
+### Manage users or apps in roles
 
-### Remove users from roles
+Depending on the type of role you choose, you will be able to assign or remove users or machine-to-machine applications in the role details page.
 
-To remove a user from the role, under the "Users" tab, click the trash bin icon in the very right of the row.
+Click the "Users" or "Machine-to-machine apps" tab to view the users or apps assigned to the role. To add a user or app to the role, click the "Assign users" or "Assign applications" button in the top right corner.
```
Comment (darcyYe) [inline docs/docs/recipes/rbac/manage-permissions-and-roles.md:56]:
```suggestion
Click the "Users" or "Machine-to-machine apps" tab to view the users or apps assigned to the role. To add user(s) or app(s) to the role, click the "Assign users" or "Assign applications" button in the top right corner.
```

**Review by darcyYe — APPROVED** [review]
Overall LGTM

File: docs/docs/recipes/rbac/manage-permissions-and-roles.md (hunk 1 above)
Comment (gao-sun) [inline docs/docs/recipes/rbac/manage-permissions-and-roles.md:31]:
```suggestion
3. A new role has two available types: "User role" or "Machine-to-machine app role".
```

**Pull request merged**